src/game_api/entities_items.hpp

```diff
@@ -739,7 +756,8 @@ class RollingItem : public Purchasable
   public:
     float roll_speed; // only positive numbers
 
-    virtual void on_purchase(Entity* who, int32_t unknown) = 0; // give you the powerup if you buy it
+    /// Skip this function for item to be unpickable
+    virtual void pickup(Entity* who, bool) = 0;
```

Comment:
Better name suggestions? (since we already have `pick_up` function for movable

Reply:
no, this is only for all the `ENT_TYPE_ITEM_PICKUP_*`, it applies the effect of giving you ropes, power up etc.
I mostly want it as hookable function, for stuff like custom pickups

Reply:
what about uh, `get_powerup`, `give_powerup`, `grant_powerup`, `pick_powerup`, `consume_powerup`, `apply_powerup`?

Reply:
I would not consider bomb bag a powerup, or like, not in current naming scheme that we or the game established
all of those suggestions sounds more like stand-alone functions rather then method on an object

Reply:
thou, if i had to i would probably choose `pick_powerup`

Reply:
> I would not consider bomb bag a powerup, or like, not in current naming scheme that we or the game established

I don't really see it like that from what I've seen, there's no POWERUP_BOMB or POWERUP_ROPE because there is just no need of such entities, and not because those aren't powerups (also I read in wikipedia, powerups like 1up or ammo are actually named powerup in there)

> all of those suggestions sounds more like stand-alone functions rather then method on an object

Fair, I would be fine with any other name as long as it's distinguishable enough from the `pick_up` functions that's just grabbing something your hands. Though for me it seems fine to me to see some code like `pickup:give_powerup(player)`, because it means giving the pickup's "contained" powerup to the player

Reply:
I guess I'll mark this as resolved now